### PR TITLE
[DOCS] Installation + Links fixed (Windows)

### PR DESCRIPTION
**Pull request by gpotter2**
As many people (https://github.com/secdev/scapy/issues/358) do not find/read the doc before running scapy, i've made the doc more obvious...

* Added missing links for Windows
* Updated old depedencies links (Windows)
* Added .bat to build the docs on a windows computer
* Some new easier installation instructions (for Windows)
* Tiny fixes in doc
* Removed useless depedencies
* Some other format fixes

### CODE REVIEW

**Comment by gpotter2** [comment]
Well i have absolutely nothing to do with this travis ci crash

**Comment by p-l-** [comment]
Thanks for that. I think @phil777 might want to host the files on secdev.org instead.

**Comment by gpotter2** [comment]
Right, but is he still active ? Cause the website dates from quite a long time ago...

**Comment by guedou** [comment]
Is dnet still mandatory ?

Concerning winpcap, I managed to successfully use [npcap](https://nmap.org/npcap/) as installed with nmap Windows installer. Being lazy, I used [chocolatey](https://chocolatey.org/) to automate the process. That's the setup that I am using to bring AppVeyor Windows CI to Scapy.

**Comment by gpotter2** [comment]
["dnet" occures](https://github.com/search?l=&q=dnet+user%3Asecdev+repo%3Ascapy&ref=advsearch&type=Code&utf8=%E2%9C%93). I don't really know... it looks that it can work without it...

So you mean we should replace pypcap with npcap ?

**Comment by guedou** [comment]
I might be wrong but Scapy on Windows only need wpcap.dll installed since winpcapy.py was integrated.

**Comment by gpotter2** [comment]
So scapy on windows definatly needs:
- pyreadline

Does it needs pywin32 ? pcap ? dnet ?

File: README.md [hunk 1]
```diff
@@ -18,7 +18,15 @@ injecting your own 802.11 frames, combining technics (VLAN hopping+ARP
 cache poisoning, VOIP decoding on WEP encrypted channel, ...),
 etc.
 
-See
+# Installation #
+
+Scapy requires a few depedencies depending of your OS to work correctly. It currently supports Unix-like operating systems (Linux, BSD, Mac OS X), and Windows.
```
Comment (guedou) [inline README.md]: I will rather reformulate this sentence as:
"Scapy works without any external Python modules on Linux and BSD like operating systems. On Windows, you need to install some mandatory dependencies as described in the [documentation](http://scapy.readthedocs.io/en/latest/installation.html).

To benefit from all Scapy features, you might want to install Python modules, such as matplotlib or pycrypto."

**Comment by guedou** [comment]
@gpotter2 on my Windows, I only installed winpcap using the nmap installer as well as the readline Python. I might be wrong but dnet does not seems mandatory.

File: README.md (hunk 1 above)
Comment (gpotter2) [inline README.md]: which is definatly better 😄

**Comment by gpotter2** [comment]
So if every one agree we might just remove the old depedencies...

**Comment by p-l-** [comment]
Yep that's supposed to be true at least ;)

**Comment by gpotter2** [comment]
Done ! I've done some other changes too...

**Comment by gpotter2** [comment]
@p-l- can you cancel this ? https://travis-ci.org/secdev/scapy/builds/184475591
Because Appveyor didn't work because of a missing import...

**Comment by p-l-** [comment]
Please **rebase** against master and remove the commit f8e44f1.

**Comment by gpotter2** [comment]
Sorry that's a branch mistake... it shouldn't be there...

**Comment by p-l-** [comment]
Can you also remove the "merge" commits (you have four of them in this PR)? Thanks!

**Comment by gpotter2** [comment]
Done (and i've fixed the conflicts)

**Comment by p-l-** [comment]
You still have one "merge" commit, or did I miss something?

File: scapy/route.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 """
 
 import socket
-from scapy.arch.consts import LOOPBACK_NAME
+from scapy.arch.consts import LOOPBACK_NAME, WINDOWS
```
Comment (p-l-) [inline scapy/route.py]: What is this exactly?

Comment (gpotter2) [inline scapy/route.py]: Oh i remember: it was in [this](https://github.com/secdev/scapy/pull/378/commits/2f04cbf737f2c86a51a50fb2141882a2f606ced7) (in another fix) and for some reason the WINDOWS was never imported... It's a tiny thing... should i remove it ?

Comment (gpotter2) [inline scapy/route.py]: Sorry it came from another fix

**Comment by gpotter2** [comment]
Removed the merge stuff + removed commit from another branch

**Comment by guedou** [comment]
I made some small comments.

@gpotter2 do you want us to [squash](https://github.com/blog/2141-squash-your-commits) (i.e. compact commits in to a single one) this PR ?

**Comment by gpotter2** [comment]
@guedou I won't mind :smiley:

File: doc/scapy/backmatter.rst [hunk 3]
```diff
@@ -4,6 +4,7 @@ Credits
 *********
 
 - Philippe Biondi is Scapy's author. He has also written most of the documentation.
+- Guillaume Valadon is one of the most active mainteners and contributors.
```
Comment (gpotter2) [inline doc/scapy/backmatter.rst]: Oh yeah sorry :/

File: README.md [hunk 4]
```diff
@@ -18,7 +18,17 @@ injecting your own 802.11 frames, combining technics (VLAN hopping+ARP
 cache poisoning, VOIP decoding on WEP encrypted channel, ...),
 etc.
 
-See
+# Installation #
+
+Scapy works without any external Python modules on Linux and BSD like operating systems. On Windows, you need to install some mandatory dependencies as described in the documentation.
+
+To benefit from all Scapy features, you might want to install Python modules, such as matplotlib or pycrypto.
```
Comment (gpotter2) [inline README.md]: There will be quite a lot of changes with the links to do in installation.rst when we'll have switched to cryptography, so I don't think we should do this now... I will do another PR once it's done

**Comment by p-l-** [comment]
@gpotter2 merge commit again...

**Comment by gpotter2** [comment]
Let me squash every thing...

File: README.md (hunk 4 above)
Comment (p-l-) [inline README.md]: It's done now, but info about cryptography can be updated in another PR.

Comment (gpotter2) [inline README.md]: It won't hurt to do it in this one i guess...

**Comment by gpotter2** [comment]
Ok i've squashed the commits and changed their names to make the PR looks cleaner.

**Comment by gpotter2** [comment]
@p-l- @guedou Sorry for being intrusive. That's just to remind you that this is still open...

File: doc/scapy/installation.rst [hunk 5]
```diff
@@ -190,18 +187,7 @@ Here are the topics involved and some examples that you can use to try if your i
...
-  
-    # pycrypto 2.6 install
-    sudo pip install pycrypto
-
-    # pycrypto 2.7a1 install
-    curl -sL https://github.com/dlitz/pycrypto/archive/v2.7a1.tar.gz | tar xz
-    cd pycrypto-2.7a1
-    python setup.py build
-    sudo python setup.py install
+* IPsec Crypto Support. ``SecurityAssociation()`` needs `Pycrypto 2.7a1 <https://github.com/dlitz/pycrypto>`_. Combined AEAD modes such as GCM and CCM are not available yet because of cryptography restrictions.
```
Comment (guedou) [inline doc/scapy/installation.rst]: I think this should refer to cryptography.

**Comment by guedou** [comment]
@gpotter2 no problem. I made a simple comment.

File: doc/scapy/installation.rst (hunk 5 above)
Comment (gpotter2) [inline doc/scapy/installation.rst]: No, because as it said "Combined AEAD modes such as GCM and CCM are not available yet because of cryptography restrictions.". We left a pycrypto part, see: https://github.com/secdev/scapy/blob/3f483a013cab1b3e59e831286d0ccb5ae69872c2/scapy/layers/ipsec.py#L164

Comment (guedou) [inline doc/scapy/installation.rst]: Sorry. I reviewed the PR way too fast =/
